Sources/Common/Communication/EventBusHandler.swift

```diff
@@ -1,66 +1,87 @@
 import Combine
 import Foundation
 
+/// Manages the handling of events through an event bus and coordinates with event storage.
+/// It handles both in-memory and file-based storage of events.
+// sourcery: InjectRegisterShared = "EventBusHandler"
+// sourcery: InjectSingleton
 public class EventBusHandler {
     private let eventBus: EventBus
     private let eventStorage: EventStorage
+    private var memoryStorage: [String: [any EventRepresentable]] = [:]
 
+    /// Initializes the EventBusHandler with an EventBus and EventStorage.
+    /// Automatically loads events from file-based storage into in-memory storage upon initialization.
     public init(eventBus: EventBus, eventStorage: EventStorage) {
         self.eventBus = eventBus
         self.eventStorage = eventStorage
+        loadEventsFromStorage()
     }
 
-    public func send<E>(_ event: E) where E: EventRepresentable {
-        if eventBus.send(event) == false {
-            let key = event.key
+    /// Loads events from file-based persistent storage into in-memory storage for each registered event type.
+    /// This facilitates quick access and ensures events can be replayed to late subscribers.
+    private func loadEventsFromStorage() {
+        EventTypesRegistry.allEventTypes().forEach { eventType in
             do {
-                try eventStorage.store(event: event, forKey: key)
+                let key = eventType.key
+                let events: [any EventRepresentable] = try eventStorage.loadEvents(ofType: eventType)
+                memoryStorage[key, default: []].append(contentsOf: events)
             } catch {
-                print("Error storing event: \(error)")
+                // Handle the error, for example, log it or take some recovery actions
+                print("Error loading events for \(eventType): \(error)")
             }
         }
     }
 
-    @discardableResult
-    public func onReceive<E: EventRepresentable>(_ eventType: E.Type, perform action: @escaping (E) -> Void) -> AnyCancellable {
-        subscribeAndReplay(eventType: eventType, action: action)
+    /// Adds an observer for a specific event type.
+    /// Replays any in-memory stored events of that type to the new observer.
+    public func addObserver<E: EventRepresentable>(_ eventType: E.Type, action: @escaping (E) -> Void) {
+        eventBus.addObserver(eventType, action: action)
+        replayEvents(forType: eventType)
```

Comment:
I am wondering if this implementation of the replay feature could cause issues 🤔. 

When thinking about the eventbus and the replay system, I was imagining that the eventbus stored an *ordered list* of events and replayed all events in order to the consumer once it attached. 

Order of events in our SDK have always mattered for the functionality of the SDK. Profile identify event has to come before tracking of events, for example. If those 2 events are delivered to consumers out of order, that could make the results going to the CDP be very different then intended. 

From reviewing the implementation of this class thus far, it seems as though the eventbus implementation does not store and replay an ordered list. 

Do you disagree that an ordered list of events is needed? 

As a bonus, I do think that 1 single list of events could make the implementation more simple in this class, too.

Reply:
I think it's different then BQ, when does an event get replyed in Event bus case? 

if an observer, subscribes to an `event`, and if there are `events` of that type stored in memory, the Event bus replays those events for the subscriber. 

So the event as the `key` its order isn't important rather the value, as in how many `IdentifyProfile` events were stored as an array value to the key is important and is being stored in order too

```
private var memoryStorage: [String: [AnyEventRepresentable]] = [:]

The order of AnyEventRepresentable instances in these arrays is preserved as per their addition.
```

Reply:
Sorry if my original message was not descriptive enough in what I would like to ask. I would like to know why order of events does not matter in the eventbus?